mesonbuild/environment.py

```diff
@@ -668,6 +734,11 @@ def move_compiler_options(properties: Properties, compopts: T.Dict[str, T.Defaul
             split_compiler_options(meson_options.host, MachineChoice.HOST)
             move_compiler_options(properties.host, compiler_options.host)
 
+        if not meson_options.host:
+            meson_options.host = None
+        if not compiler_options.host:
+            compiler_options.host = None
```

Comment:
If you don't initialize the `PerMachineDefaultable` above, you won't need this.

Reply:
This allows us to delete a ton of:
```python
if compiler_options.host is None:
    compiler_options.host = defaultdict(dict)
```

Which I think is nice.